### DISCUSSION TITLE
API Query Row Limit

### DISCUSSION THREAD

**Discussion by Kuuchuu** [nocodb/nocodb#900] [Q&A] [answered]
When I am querying my database the API only returns 100 rows. I tried modifying the limit parameter but it seemed to cap at 100. Is this a hardcoded limit? We have over 600 rows added a day as business transactions. We are needing to pull every row that has the current business date. Any help/guidance would be greatly appreciated! Thanks

**Comment by wingkwong** [comment, marked as answer]
The default limit is calculated as ``max(min(DB_QUERY_LIMIT_DEFAULT , DB_QUERY_LIMIT_MAX), DB_QUERY_LIMIT_MIN)``. By default the initial values for ``DB_QUERY_LIMIT_DEFAULT``, ``DB_QUERY_LIMIT_MAX``, and ``DB_QUERY_LIMIT_MIN`` are ``25``, ``100``, and ``1`` respectively. 

Therefore, you need to increase both ``DB_QUERY_LIMIT_DEFAULT`` and ``DB_QUERY_LIMIT_MAX``.

**Comment by o1lab** [comment]
I suggest to keep the limit to 100 and use pagination ([limit and offset](https://docs.nocodb.com/developer-resources/rest-apis#query-params)) in APIs. 

This is since the querying can also involve joins and can be expensive. Some of our users have 100s of columns and 10s of them being joined.

Use it as per your needs.
